### PR TITLE
AUD-947 Truncate display names on signup if necessary

### PR DESCRIPTION
**Pull request by rickyrombo**
### Description

- Fixes bug causing Instagram signup flow to not autofill for all desktop users
- Fixes bug when Twitter/Instagram names are longer than 32 characters, asks user to verify the truncated names.

### Dragons

Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.

- Tested locally by placing breakpoints in `TwitterOverlay`'s `onClickInstagram` function, and then in the console running one of the following to fake a login from instagram or twitter respectively: 
```js
setTimeout( () => props.onInstagramLogin('', { username: 'foo', full_name:'abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyz', verified: false, id: 'idk'}), 5000)
```
```js
setTimeout( () => props.onTwitterLogin(new Response(JSON.stringify({uuid: '', profile: { screen_name: 'foo', name:'abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyz', verified: false, profile_image_url_https: 'https://pbs.twimg.com/app_img/1112838394780868609/A7KVqmDi?format=png&name=73x73'}}))), 5000)
```

### How will this change be monitored?

For features that are critical or could fail silently please describe the monitoring/alerting being added.

- None.

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://frontend-demo.audius.co/mjp-truncate-display-names

File: src/containers/sign-on/utils/formatSocialProfile.ts [hunk 1]
```diff
@@ -4,6 +4,7 @@ import { resizeImage } from 'utils/imageProcessingUtil'
 const GENERAL_ADMISSION = process.env.REACT_APP_GENERAL_ADMISSION
 
 export const MAX_HANDLE_LENGTH = 16
+export const MAX_DISPLAY_NAME_LENGTH = 32
```
Comment (raymondjacobson) [inline src/containers/sign-on/utils/formatSocialProfile.ts:7]: can we share this const?
https://github.com/AudiusProject/audius-client/blob/7a4a87b0cc6d89cb6d5478223dc421e609b4542e/src/containers/sign-on/components/ProfileForm.js#L88

File: src/containers/sign-on/utils/formatSocialProfile.ts [hunk 2]
```diff
@@ -83,6 +88,13 @@ export const formatInstagramProfile = async (
       )
     }
   }
+  if ((instagramProfile.full_name?.length || 0) > MAX_DISPLAY_NAME_LENGTH) {
```
Comment (raymondjacobson) [inline src/containers/sign-on/utils/formatSocialProfile.ts:91]: in the case we have no full_name, wouldn't we want the user to review? (not sure if this happens in practice). i think the check maybe should be

`if (!instagramProfile.full_name || instagramProfile.full_name.length > MAX) {`

**Review by raymondjacobson — APPROVED** [review]

**Comment by audius-infra** [comment]
Preview this change https://frontend-demo.audius.co/mjp-truncate-display-names

**Pull request merged**